doc/developers/contributing.rst

```diff
@@ -210,37 +207,25 @@ then submit a "pull request" (PR):
        $ git clone git@github.com:YourLogin/scikit-learn.git
        $ cd scikit-learn
 
-4. Install library in editable mode::
+4. Install the development dependencies::
+
+       $ pip install numpy scipy cython pytest flake8
+
+5. Install scikit-learn in editable mode::
 
        $ pip install --editable .
 
    for more details about advanced installation, see the
    :ref:`install_bleeding_edge` section.
 
-5. Create a branch to hold your development changes::
+6. Create a branch to hold your development changes::
 
        $ git checkout -b my-feature
 
    and start making changes. Always use a ``feature`` branch. It's good practice to
    never work on the ``master`` branch!
 
-   .. note::
-
-     In the above setup, your ``origin`` remote repository points to
-     ``YourLogin/scikit-learn.git``. If you wish to fetch/merge from the main
-     repository instead of your forked one, you will need to add another remote
-     to use instead of ``origin``. If we choose the name ``upstream`` for it, the
-     command will be::
-
-         $ git remote add upstream https://github.com/scikit-learn/scikit-learn.git
-
-     And in order to fetch the new remote and base your work on the latest changes
-     of it you can::
-
-         $ git fetch upstream
-         $ git checkout -b my-feature upstream/master
```

Comment:
I like the re-ordering. But I think in the PR we added these lines, we discussed that users should _always_ do `git checkout -b my-feature upstream/master` for a new branch, which I think is not missing in the docs with your proposed changes.

Reply:
Ok. I'll add the addition of the upstream ref as a separate step.